### PR TITLE
feat: add hook for collecting recently visited pages

### PR DESCRIPTION
**Pull request by daveleek**
## About the changes

Adds a hook that captures and makes available a list of 5 most recent visited pages/features/projects

### CODE REVIEW

File: frontend/src/hooks/useRecentlyVisited.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    const location = useLocation();
+
+    useEffect(() => {
+        if (!location.pathname) return;
+
+        const path = routes.find((r) => r.path === location.pathname);
+        if (path) {
+            setCappedLastVisited([{ pathName: path.path }]);
+        } else {
+            if (featureMatch?.params.featureId) {
```
Comment (sjaanus) [inline frontend/src/hooks/useRecentlyVisited.ts:56]: We can use just else if

File: frontend/src/hooks/useRecentlyVisited.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+    useEffect(() => {
+        if (lastVisited) {
+            setLocalStorageItem(key, lastVisited);
+            emitEvent();
+        }
+    }, [JSON.stringify(lastVisited), key, emitEvent]);
+
+    const setCappedLastVisited = useCallback(
+        (pages: LastViewedPage[]) => {
```
Comment (sjaanus) [inline frontend/src/hooks/useRecentlyVisited.ts:79]: This is probably single page.

File: frontend/src/hooks/useRecentlyVisited.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            )
+                return;
+            if (
+                lastVisited.find(
+                    (item) => item.pathName && item.pathName === page.pathName,
+                )
+            )
+                return;
+            if (
+                lastVisited.find(
```
Comment (sjaanus) [inline frontend/src/hooks/useRecentlyVisited.ts:81]: Can we extract a const, so it describes what this actually does?

File: frontend/src/hooks/useRecentlyVisited.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        (page: LastViewedPage) => {
+            if (page.featureId && !page.projectId) return;
+            if (
+                lastVisited.find(
+                    (item) =>
+                        item.featureId && item.featureId === page.featureId,
+                )
+            )
+                return;
+            if (
```
Comment (sjaanus) [inline frontend/src/hooks/useRecentlyVisited.ts:74]: I wonder how difficult would it to bring it back to top1? Instead of ignoring.

**Comment by daveleek** [comment]
Merged as part of Command bar PoC [PR](https://github.com/Unleash/unleash/pull/7350)

**Pull request closed without merging**